soh/soh/Enhancements/randomizer/randomizer.h

```diff
@@ -12,6 +12,136 @@
 #define NUM_NAVI_MESSAGES 19
 #define NUM_ICE_TRAP_MESSAGES 23
 
+typedef enum RandoOption {
+  //Generic Settings (any binary option can use this)
+  //(off/on, no/yes, don't skip/skip)
+  RO_GENERIC_OFF,
+  RO_GENERIC_ON,
+  RO_GENERIC_NO = 0,
+  RO_GENERIC_YES,
+  RO_GENERIC_DONT_SKIP = 0,
+  RO_GENERIC_SKIP,
+
+  //Forest settings (closed, closed deku, open)
+  RO_FOREST_CLOSED = 0,
+  RO_FOREST_CLOSED_DEKU,
+  RO_FOREST_OPEN,
+
+  //Door of Time settings (open, song only, closed)
+  RO_DOOROFTIME_OPEN = 0,
+  RO_DOOROFTIME_SONGONLY,
+  RO_DOOROFTIME_CLOSED,
+  
+  //Zora's Fountain settings (closed, closed as child, open)
+  RO_ZF_CLOSED = 0,
+  RO_ZF_CLOSED_CHILD,
+  RO_ZF_OPEN,
+
+  //Starting Age settings (child, adult, random)
+  RO_AGE_CHILD = 0,
+  RO_AGE_ADULT,
+  RO_AGE_RANDOM,
+
+  //Gerudo Fortress settings (normal, fast, open)
+  RO_GF_NORMAL = 0,
+  RO_GF_FAST,
+  RO_GF_OPEN,
+
+  //Rainbow Bridge settings (always open, vanilla, stones, medallions, dungeon rewards, dungeons, tokens)
+  RO_BRIDGE_ALWAYS_OPEN = 0,
+  RO_BRIDGE_VANILLA,
+  RO_BRIDGE_STONES,
+  RO_BRIDGE_MEDALLIONS,
+  RO_BRIDGE_DUNGEON_REWARDS,
+  RO_BRIDGE_DUNGEONS,
+  RO_BRIDGE_TOKENS,
+
+  //Shopsanity settings (off, 0-4 items, random)
+  RO_SHOPSANITY_OFF = 0,
+  RO_SHOPSANITY_ZERO_ITEMS,
+  RO_SHOPSANITY_ONE_ITEM,
+  RO_SHOPSANITY_TWO_ITEMS,
+  RO_SHOPSANITY_THREE_ITEMS,
+  RO_SHOPSANITY_FOUR_ITEMS,
+  RO_SHOPSANITY_RANDOM,
+
+  //Scrubsanity settings (off, affordable, expensive, random)
+  RO_SCRUBS_OFF = 0,
+  RO_SCRUBS_AFFORDABLE,
+  RO_SCRUBS_EXPENSIVE,
+  RO_SCRUBS_RANDOM,
+
+  //Shuffle Merchants settings (off, on w/o hints, on w/hints)
+  RO_MERCHANTS_OFF = 0,
+  RO_MERCHANTS_ON_NO_HINTS,
+  RO_MERCHANTS_ON_WITH_HINTS,
+
+  //Bombchu settings (on, "on+bombchu", off)
+  RO_BOMBCHU_DROPS_ON = 0,
+  RO_BOMBCHU_DROPS_ON_PLUS,
+  RO_BOMBCHU_DROPS_OFF,
```

Comment:
These are for item drops in general, not just Bombchu

Reply:
```suggestion
  // Ammo drop settings (on, "on+bombchu", off)
  RO_AMMO_DROPS_ON = 0,
  RO_AMMO_DROPS_ON_PLUS_BOMBCHU,
  RO_AMMO_DROPS_OFF,
```

maybe ammo drops is a good name? maybe ammo isn't the best name because of recovery hearts and rupees?

Reply:
Just double checked and it is just ammo drops in 3D rando, so that'll work.